### DISCUSSION TITLE
Writing Arduino .ino files

### DISCUSSION THREAD

**Discussion by twheatcroft07** [bonsai-rx/bonsai#1581] [Q&A]
Hi there,

I'm wondering whether I can get bonsai to automatically save the Arduino program I'm using, along with the data from the session. I don't know of a way for Bonsai to copy files from place to place, or for bonsai to read in .ino files. I could convert the Arduino files to .csv files, and then have Bonsai read in that csv, and then write a new one, but I was wondering if there's an alternative.

Is there any way of achieving this?

Thanks,
Tom

**Comment by bruno-f-cruz** [comment]
Check https://github.com/orgs/bonsai-rx/discussions/1002

You can also just use `Read`/`WriteAllText` operators.
